README.md

```diff
@@ -8,131 +8,133 @@ products:
-2. Microsoft.Azure.KeyVault - contains methods for interacting with Key Vault. 
+*  Go to the [Azure Portal] and log in using your Azure account
+*  Search for your Key Vault in **Search Resources dialog box**
+*  Select **Overview** > **Access policies**
+*  Click on **Add Access Policy** > **Secret permissions** > **Get**
+*  Click on **Select Principal**, add your account and pre created **system-assigned identity**
+*  Click on "OK" to add the new Access Policy, then click "Save" to save the Access Policy
 
-The relevant code is in WebAppKeyVault/WebAppKeyVault/Controllers/HomeController.cs file. The **AzureServiceTokenProvider** class (which is part of Microsoft.Azure.Services.AppAuthentication) tries the following methods to get an access token:-
-1. Managed Service Identity (MSI) - for scenarios where the code is deployed to Azure, and the Azure resource supports MSI. 
-2. [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) (for local development) - Azure CLI version 2.0.12 and above supports the **get-access-token** option. AzureServiceTokenProvider uses this option to get an access token for local development. 
-3. Active Directory Integrated Authentication (for local development). To use integrated Windows authentication, your domain’s Active Directory must be federated with Azure Active Directory. Your application must be running on a domain-joined machine under a user’s domain credentials.
+#### Step 2: Copy and save Key Vault Url.
 
-```csharp    
-public async System.Threading.Tasks.Task<ActionResult> Index()
-{
-    AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+Select **Overview** > **DNS Name**, copy the associated **Key Vault Url** to the clipboard, then paste it into a text editor for later use.
 
-    try
-    {
-        var keyVaultClient = new KeyVaultClient(
-            new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
+## Run the application
+Clone the repo to your development machine. 
 
-        var secret = await keyVaultClient.GetSecretAsync("https://keyvaultname.vault.azure.net/secrets/secret")
-            .ConfigureAwait(false);
+```bash
+git clone https://github.com/Azure-Samples/app-service-msi-keyvault-dotnet.git
+```
 
-        ViewBag.Secret = $"Secret: {secret.Value}";
-        
-    }
-    catch (Exception exp)
-    {
-        ViewBag.Error = $"Something went wrong: {exp.Message}";
-    }
+### Run the application on your local development machine
+To run the sample, this solution requires a Key Vault Url is stored in an environment variable on the machine , and [register an application with the Microsoft identity platform],
+then grant the access policy by [Step 1: Set access policy].
 
-    ViewBag.Principal = azureServiceTokenProvider.PrincipalUsed != null ? $"Principal Used: {azureServiceTokenProvider.PrincipalUsed}" : string.Empty;
+Linux
```

Comment:
I'd make these subheadings, like "###". That way, they stand out and can be linked. Would be more consistent with other documentation I've seen like this.

Reply:
Resovled